aiohue/sensors.py

```diff
@@ -53,3 +71,367 @@ def state(self):
+        return self._daylight
+
+    @property
+    def on(self):
+        return self._on
+
+    async def set_config(self, on=None, long=None, lat=None,
+                         sunriseoffset=None, sunsetoffset=None):
+        """Change config of a Daylight sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+                'long': long,
+                'lat': lat,
+                'sunriseoffset': sunriseoffset,
+                'sunsetoffset': sunsetoffset,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class CLIPPresenceSensor(GenericCLIPSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._presence = raw['state'].get('presence')
+
+    @property
+    def presence(self):
+        return self._presence
+
+    async def set_config(self, on=None):
+        """Change config of a CLIP Presence sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class ZLLPresenceSensor(GenericZLLSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._presence = raw['state'].get('presence')
+
+    @property
+    def presence(self):
+        return self._presence
+
+    async def set_config(self, on=None, sensitivity=None, sensitivitymax=None):
+        """Change config of a ZLL Presence sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+                'sensitivity': sensitivity,
+                'sensitivitymax': sensitivitymax,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class CLIPSwitchSensor(GenericCLIPSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._buttonevent = raw['state'].get('buttonevent')
+
+    def buttonevent(self):
+        return self._buttonevent
+
+    async def set_config(self, on=None):
+        """Change config of a CLIP Switch sensor."""
+        data = {
+            key: value for key, value in {
+                'on': on,
+            }.items() if value is not None
+        }
+
+        await self._request('put', 'sensors/{}/config'.format(self.id),
+                            json=data)
+
+
+class ZGPSwitchSensor(GenericZGPSensor):
+    def __init__(self, id, raw, request):
+        super().__init__(id, raw, request)
+        self._buttonevent = raw['state'].get('buttonevent')
+
+    @property
+    def buttonevent(self):
+        if self._buttonevent == 34:
+            return 'BUTTON_1'
```

Comment:
We should not convert and then we should expose constants `ZGP_SWITCH_BUTTON_1` etc.

That way we can do:

```python
from aiohue.sensors import ZGP_SWITCH_BUTTON_1

if sensor.buttonevent == ZGP_SWITCH_BUTTON_1:
  …

```